service/resource/dnsrecord/resource.go

```diff
@@ -0,0 +1,351 @@
+package dnsrecord
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/arm/dns"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/giantswarm/azuretpr"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/framework"
+
+	"github.com/giantswarm/azure-operator/client"
+	"github.com/giantswarm/azure-operator/service/key"
+)
+
+const (
+	// Name is the identifier of the resource.
+	Name = "resourcegroup"
+)
+
+// Config is the resource group Resource configuration.
+type Config struct {
+	// Dependencies.
+
+	AzureConfig client.AzureConfig
+	Logger      micrologger.Logger
+}
+
+// DefaultConfig provides a default configuration to create a new resource by
+// best effort.
+func DefaultConfig() Config {
+	return Config{
+		// Dependencies.
+		AzureConfig: client.DefaultAzureConfig(),
+		Logger:      nil,
+	}
+}
+
+// Resource manages Azure resource groups.
+type Resource struct {
+	// Dependencies.
+
+	azureConfig client.AzureConfig
+	logger      micrologger.Logger
+}
+
+// New creates a new configured resource group resource.
+func New(config Config) (*Resource, error) {
+	// Dependencies.
+	if err := config.AzureConfig.Validate(); err != nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.AzureConfig.%s", err)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Logger must not be empty")
+	}
+
+	newService := &Resource{
+		azureConfig: config.AzureConfig,
+		logger: config.Logger.With(
+			"resource", Name,
+		),
+	}
+
+	return newService, nil
+}
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+	o, err := toCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return r.getCurrentState(ctx, o)
+}
```

Comment:
The other operators define this in different files. I thought we already had consensus on this practise.

Reply:
Again copy/paste. I'm not big on that, but will change.

Reply:
Thanks.